File: yangxj96-starter/yangxj96-starter-security/src/main/java/com/yangxj96/saas/starter/security/config/SaLogForSlf4j.java
package com.yangxj96.saas.starter.security.config;

import cn.dev33.satoken.log.SaLog;
import lombok.extern.slf4j.Slf4j;

/**
 * 日志转接
 */
@Slf4j
public class SaLogForSlf4j implements SaLog {


    SaLogForSlf4j() {
        log.atDebug().log("${EnvCons.PREFIX}注入satoken的打印配置");
    }

    @Override
    public void trace(String str, Object... args) {
        log.trace(str, args);
    }

    @Override
    public void debug(String str, Object... args) {
        log.debug(str, args);
    }

    @Override
    public void info(String str, Object... args) {
        log.info(str, args);
    }

    @Override
    public void warn(String str, Object... args) {
        log.warn(str, args);
    }

    @Override
    public void error(String str, Object... args) {
        log.error(str, args);
    }

    @Override
    public void fatal(String str, Object... args) {
        log.error(str, args);
    }
}

File: yangxj96-starter/yangxj96-starter-security/src/main/kotlin/com/yangxj96/saas/starter/security/config/SaLogForSlf4j.kt
package com.yangxj96.saas.starter.security.config

import cn.dev33.satoken.log.SaLog
import com.yangxj96.saas.starter.security.constant.EnvCons
import org.slf4j.LoggerFactory

/**
 * 日志转接
 */
class SaLogForSlf4j : SaLog {

    companion object {
        private val log = LoggerFactory.getLogger(this::class.java)
    }

    init {
        log.atDebug().log("${EnvCons.PREFIX}注入satoken的打印配置")
    }

    override fun trace(str: String?, vararg args: Any?) {
        log.trace(str, args)
    }

    override fun debug(str: String?, vararg args: Any?) {
        log.debug(str, args)
    }

    override fun info(str: String?, vararg args: Any?) {
        log.info(str, args)
    }

    override fun warn(str: String?, vararg args: Any?) {
        log.warn(str, args)
    }

    override fun error(str: String?, vararg args: Any?) {
        log.error(str, args)
    }

    override fun fatal(str: String?, vararg args: Any?) {
        log.error(str, args)
    }
}